src/client/src/tscLocal.c

```diff
@@ -295,7 +295,7 @@ void tscSCreateCallBack(void *param, TAOS_RES *tres, int code) {
     taos_fetch_rows_a(tres, tscSCreateCallBack, param);    
     builder->callStage = SCREATE_CALLBACK_RETRIEVE;
   } else {
-    char *result = calloc(1, TSDB_MAX_BINARY_LEN);
+    char *result = calloc(1, TSDB_MAX_SQL_LEN);
```

Comment:
Use TSDB_MAX_SQL_LEN to control the length of rebuilt sql stmt, instead of TSDB_MAX_BINARY_LEN.